### PR TITLE
[front] - enh(AssistantPicker): focus on searchbar

### PR DESCRIPTION
**Pull request by JulesBelveze**
## Description

This PR aims at enhancing the `AssistantPicker` component to focus on the search bar when the dropdown opens.

**Reference:**
- https://github.com/dust-tt/tasks/issues/1254

## Risk

None

## Deploy Plan

Deploy `front`

### CODE REVIEW

**Review by tdraier — APPROVED** [review]

File: front/components/assistant/AssistantPicker.tsx [hunk 1]
```diff
@@ -44,6 +44,14 @@ export function AssistantPicker({
     setSearchedAssistants(filterAndSortAgents(assistants, searchText));
   }, [searchText, assistants]);
 
+  const searchbarRef = (element: HTMLInputElement) => {
```
Comment (flvndvd) [inline front/components/assistant/AssistantPicker.tsx:47]: If the delay is here to ensure the component is mounted, there is a more elegant solution that consist of using `useEffect` with an empty array to ensure it's called only once after the component has been mounted.

Comment (JulesBelveze) [reply front/components/assistant/AssistantPicker.tsx:47]: The delay is mainly here to make sure the inputbar doesn't get the focus

Comment (JulesBelveze) [reply front/components/assistant/AssistantPicker.tsx:47]: We tried your `useEffect` approach but didn't work as the input bar was getting the focus

Comment (flvndvd) [reply front/components/assistant/AssistantPicker.tsx:47]: 👀

Comment (flvndvd) [reply front/components/assistant/AssistantPicker.tsx:47]: Then I guess the issue is that we have some race condition with a code that triggers the focus on the input bar, no?

Comment (flvndvd) [reply front/components/assistant/AssistantPicker.tsx:47]: Basically you are fighting this guy: https://github.com/dust-tt/dust/blob/3b80285dbbd7df463df30a6922601f05e1c0ef9e/front/components/assistant/conversation/input_bar/InputBarContainer.tsx#L80-L84

Comment (JulesBelveze) [reply front/components/assistant/AssistantPicker.tsx:47]: To overcome this I guess we would need to set focus on the inputbar if we click anywhere but the 3 action buttons? 🥵

**Pull request merged**